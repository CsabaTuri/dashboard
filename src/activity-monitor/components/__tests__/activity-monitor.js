import React from "react"
import renderer from "react-test-renderer"

import { reduceActivityMonitor, monitorActivity, ActivityMonitor } from "../../"

describe("ActivityMonitor Component", () => {
  let comp, store

  beforeEach(() => {
    store = createImmutableTestStore({ activityMonitor: reduceActivityMonitor })()
    comp = renderer.create(<ActivityMonitor store={store} />)
    expect(comp).toMatchSnapshot()
  })

  it("should turn spinner on and off when monitoring activity", () => {
    return store.dispatch(
      monitorActivity(() => {
        expect(comp).toMatchSnapshot() // Inside monitor. Display spinner
        return Promise.resolve()
      })
    ).then(() => {
      expect(comp).toMatchSnapshot() // Outside monitor. Hide spinner
    })
  })
})
