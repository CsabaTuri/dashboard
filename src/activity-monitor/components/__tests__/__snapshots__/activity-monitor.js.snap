exports[`test should turn spinner OFF when NOT monitoring activity 1`] = `null`;

exports[`test should turn spinner ON when monitoring activity 1`] = `
<Spinner
  className="activity-monitor"
  type={0} />
`;
